src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -13,318 +13,28 @@
-           opt == DispatchTypes::GraphicsOptions::Steady ||
-           opt == DispatchTypes::GraphicsOptions::Invisible ||
-           opt == DispatchTypes::GraphicsOptions::Visible ||
-           opt == DispatchTypes::GraphicsOptions::CrossedOut ||
-           opt == DispatchTypes::GraphicsOptions::NotCrossedOut;
-}
-
-// Routine Description:
-// Returns true if the GraphicsOption represents an extended color option.
-//   These are followed by up to 4 more values which compose the entire option.
-// Return Value:
-// - true if the opt is the indicator for an extended color sequence, false otherwise.
-static constexpr bool _isRgbColorOption(const DispatchTypes::GraphicsOptions opt) noexcept
-{
-    return opt == DispatchTypes::GraphicsOptions::ForegroundExtended ||
-           opt == DispatchTypes::GraphicsOptions::BackgroundExtended;
-}
-
-// Routine Description:
-// Returns true if the GraphicsOption represents an extended color option.
-//   These are followed by up to 4 more values which compose the entire option.
-// Return Value:
-// - true if the opt is the indicator for an extended color sequence, false otherwise.
-static constexpr bool _isBoldColorOption(const DispatchTypes::GraphicsOptions opt) noexcept
-{
-    return opt == DispatchTypes::GraphicsOptions::BoldBright ||
-           opt == DispatchTypes::GraphicsOptions::UnBold;
-}
-
-// Function Description:
-// - checks if this graphics option should set either the console's FG or BG to
-//the default attributes.
-// Return Value:
-// - true if the opt sets either/or attribute to the defaults, false otherwise.
-static constexpr bool _isDefaultColorOption(const DispatchTypes::GraphicsOptions opt) noexcept
-{
-    return opt == DispatchTypes::GraphicsOptions::Off ||
-           opt == DispatchTypes::GraphicsOptions::ForegroundDefault ||
-           opt == DispatchTypes::GraphicsOptions::BackgroundDefault;
-}
-
-#pragma warning(pop)
+// clang-format off
+const BYTE BLUE_ATTR      = 0x01;
+const BYTE GREEN_ATTR     = 0x02;
+const BYTE RED_ATTR       = 0x04;
+const BYTE BRIGHT_ATTR    = 0x08;
+const BYTE DARK_BLACK     = 0;
+const BYTE DARK_RED       = RED_ATTR;
+const BYTE DARK_GREEN     = GREEN_ATTR;
+const BYTE DARK_YELLOW    = RED_ATTR | GREEN_ATTR;
+const BYTE DARK_BLUE      = BLUE_ATTR;
+const BYTE DARK_MAGENTA   = RED_ATTR | BLUE_ATTR;
+const BYTE DARK_CYAN      = GREEN_ATTR | BLUE_ATTR;
+const BYTE DARK_WHITE     = RED_ATTR | GREEN_ATTR | BLUE_ATTR;
+const BYTE BRIGHT_BLACK   = BRIGHT_ATTR;
+const BYTE BRIGHT_RED     = BRIGHT_ATTR | RED_ATTR;
+const BYTE BRIGHT_GREEN   = BRIGHT_ATTR | GREEN_ATTR;
+const BYTE BRIGHT_YELLOW  = BRIGHT_ATTR | RED_ATTR | GREEN_ATTR;
+const BYTE BRIGHT_BLUE    = BRIGHT_ATTR | BLUE_ATTR;
+const BYTE BRIGHT_MAGENTA = BRIGHT_ATTR | RED_ATTR | BLUE_ATTR;
+const BYTE BRIGHT_CYAN    = BRIGHT_ATTR | GREEN_ATTR | BLUE_ATTR;
+const BYTE BRIGHT_WHITE   = BRIGHT_ATTR | RED_ATTR | GREEN_ATTR | BLUE_ATTR;

```

Comment:
we can probably call these `static constexpr`; static so they don't appears as global symbols, and constexpr because it feels most right

Reply:
Again I'm in agreement with you on this - I prefer `constexpr` - but I just copied the format from the `TerminalDispatchGraphics` implementation for consistency (see [here](https://github.com/microsoft/terminal/blob/d711d731d7d00259b2e0fe5847928c9f1c1121ed/src/cascadia/TerminalCore/TerminalDispatchGraphics.cpp#L9-L30)). And note that both `const` and `constexpr` have internal linkage in C++ (at least by default), so `static` shouldn't be necessary as I understand it.

Also, I'm rather hoping we can rid of these constants in the long term anyway. Or at least unify the two sets into a shared header somewhere. It's just that we've got this weird situation at the moment where the conhost color table is in a different order to the one in Windows Terminal. Ideally we'll get those in sync at some point.

Reply:
> `const` and `constexpr` have internal linkage

Fine by me! I'm all for "unify these later and clean them up at the same time". THanks!

Reply:
We do have `src/inc/conattrs.h`, which might be an appropriate place for them 🤔?

Reply:
Problem is the constants in `AdaptDispatch` are currently different from the ones in `TerminalDispatch`. Once we can unify them, assuming that's possible, I'd be happy to move them to conattrs. I wanted to leave that for a later PR, though, because I think it might be complicated, and it's not strictly necessary to get to #2661.